upgrade/util/util.go

```diff
@@ -144,3 +148,770 @@ func CreateOrUpdateLonghornVersionSetting(namespace string, lhClient *lhclientse
+	ims, err := ListInstanceManagers(namespace, lhClient, resourceMaps)
+	if err != nil {
+		return nil, err
+	}
+	im, ok := ims[name]
+	if !ok {
+		return nil, apierrors.NewNotFound(v1beta2.Resource("instancemanager"), name)
+	}
+	return im, nil
+}
+
+func GetShareManager(namespace string, lhClient *lhclientset.Clientset, resourceMaps map[string]interface{}, name string) (*v1beta2.ShareManager, error) {
+	sms, err := ListShareManagers(namespace, lhClient, resourceMaps)
+	if err != nil {
+		return nil, err
+	}
+	sm, ok := sms[name]
+	if !ok {
+		return nil, apierrors.NewNotFound(v1beta2.Resource("sharemanager"), name)
+	}
+	return sm, nil
+}
+
+func GetVolume(namespace string, lhClient *lhclientset.Clientset, resourceMaps map[string]interface{}, name string) (*v1beta2.Volume, error) {
+	volumes, err := ListVolumes(namespace, lhClient, resourceMaps)
+	if err != nil {
+		return nil, err
+	}
+	volume, ok := volumes[name]
+	if !ok {
+		return nil, apierrors.NewNotFound(v1beta2.Resource("volume"), name)
+	}
+	return volume, nil
+}
+
+func GetBackingImage(namespace string, lhClient *lhclientset.Clientset, resourceMaps map[string]interface{}, name string) (*v1beta2.BackingImage, error) {
+	bis, err := ListBackingImages(namespace, lhClient, resourceMaps)
+	if err != nil {
+		return nil, err
+	}
+	bi, ok := bis[name]
+	if !ok {
+		return nil, apierrors.NewNotFound(v1beta2.Resource("backingimage"), name)
+	}
+	return bi, nil
+}
+
+func GetBackingImageDataSource(namespace string, lhClient *lhclientset.Clientset, resourceMaps map[string]interface{}, name string) (*v1beta2.BackingImageDataSource, error) {
+	bidss, err := ListBackingImageDataSources(namespace, lhClient, resourceMaps)
+	if err != nil {
+		return nil, err
+	}
+	bids, ok := bidss[name]
+	if !ok {
+		return nil, apierrors.NewNotFound(v1beta2.Resource("backingimagedatasource"), name)
+	}
+	return bids, nil
+}
+
+func GetRecurringJob(namespace string, lhClient *lhclientset.Clientset, resourceMaps map[string]interface{}, name string) (*v1beta2.RecurringJob, error) {
+	recurringJobs, err := ListRecurringJobs(namespace, lhClient, resourceMaps)
+	if err != nil {
+		return nil, err
+	}
+	recurringJob, ok := recurringJobs[name]
+	if !ok {
+		return nil, apierrors.NewNotFound(v1beta2.Resource("recurringjob"), name)
+	}
+	return recurringJob, nil
+}
+
+func CreateRecurringJob(namespace string, lhClient *lhclientset.Clientset, resourceMaps map[string]interface{}, job *longhorn.RecurringJob) (*v1beta2.RecurringJob, error) {
+	obj, err := lhClient.LonghornV1beta2().RecurringJobs(namespace).Create(context.TODO(), job, metav1.CreateOptions{})
+	if err != nil {
+		return obj, err
+	}
+
+	var recurringJobs map[string]*v1beta2.RecurringJob
+	if v, ok := resourceMaps[types.LonghornKindRecurringJob]; ok {
+		recurringJobs = v.(map[string]*v1beta2.RecurringJob)
+	} else {
+		recurringJobs = map[string]*v1beta2.RecurringJob{}
```

Comment:
For this case, do we need `resourceMaps[types.LonghornKindRecurringJob] = recurringJobs`?
Similar below

Reply:
For `recurringJobs`, it's not really necessary, because `recurringJobs` are GET and CREATE. `resourceMaps[types.LonghornKindRecurringJob]` is just for consistency. We can remove it.

Reply:
I mean, if `resourceMaps[types.LonghornKindRecurringJob]` does not exist, should we add this newly created RecurringJob map to `resourceMaps`?

Reply:
I see. From the logics related to `RecurringJob`, it should be fine and not see any issue for now.

Reply:
Here all creation functions have similar formats. I would suggest adding this handling for all of them.

Reply:
> Here all creation functions have similar formats. I would suggest adding this handling for all of them.

Sorry. What's the handling?

Reply:
cc @shuo-wu

Reply:
I mean, adding something like `resourceMaps[types.LonghornKindRecurringJob] = recurringJobs` for all creation functions.

Reply:
Updated. Thank you.